### PR TITLE
Edit Reference guide with `faas-cli secret`

### PR DESCRIPTION
**Pull request by ivanayov**
Edit secrets section within the Reference guide to document usage
of the nesw Secrets API and CLI command

Signed-off-by: Ivana Yovcheva (VMware) <iyovcheva@vmware.com>

## Motivation and Context

- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))
Raised on slack

## How Has This Been Tested?

By running docs locally:

screen shot 2019-01-25 at 16 13 32

and 
screen shot 2019-01-25 at 16 14 27

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`

### CODE REVIEW

File: docs/reference/secrets.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ Using secrets is a two step process. First you need to define a new secret in yo
 * Secrets can be specified via API, CLI or YAML file
 * You can use one to many secrets in a function
 * Secrets must exist in the cluster at deployment time
-* Secrets need to be created with `kubectl` or `docker secret create`, but [in the near future](https://github.com/openfaas/faas/issues/807) an API will exist to create, list, delete and update secrets.
+* Secrets need to be created with `faas-cli secret create`
```
[the remark this answers is not in the document]
Comment (ivanayov) [reply docs/reference/secrets.md]: Thanks @rgee0

File: docs/reference/secrets.md [hunk 2]
```diff
@@ -8,8 +8,8 @@ Using secrets is a two step process. First you need to define a new secret in yo
 
 * Secrets can be specified via API, CLI or YAML file
 * You can use one to many secrets in a function
-* Secrets must exist in the cluster at deployment time
-* Secrets need to be created with `kubectl` or `docker secret create`, but [in the near future](https://github.com/openfaas/faas/issues/807) an API will exist to create, list, delete and update secrets.
+* Specified secrets must exist in the cluster at function deploy. A function may consume zero to many secrets
```
Comment (alexellis) [inline docs/reference/secrets.md]: "at function deploy" isn't correct, should be "at the function's deployment time"

Comment (alexellis) [reply docs/reference/secrets.md]: >  A function may consume zero to many secrets

not needed, it's said above

File: docs/reference/secrets.md [hunk 3]
```diff
@@ -31,44 +31,9 @@ R^YqzKzSJw51K9zPpQ3R3N
 
 Now we can import the secret into the cluster.
 
-#### Define a secret in Kubernetes
```
Comment (alexellis) [inline docs/reference/secrets.md]: Should we really delete this or just add it as an advanced section instead?

**Comment by alexellis** [comment]
Hopefully I covered everything in 
https://github.com/openfaas/docs/commit/b727c19d57cae2f447e7c48112d0eb97d8548994

**Pull request closed without merging**